From c3fa17b055089e07266053101c7fc3f86d72e293 Mon Sep 17 00:00:00 2001
From: Chuanxiao Dong <chuanxiao.dong@intel.com>
Date: Wed, 19 Sep 2012 15:21:54 +0800
Subject: [PORT FROM MAIN] vold: remove "-c 8" formatting params

BZ: 57307

Remove "-c 8" formatting params. Let the formatting tool to determine
the block size automatically regarding the device size. This can improve
the formatting time, as well as some random/sequential performance

Change-Id: I905bd8a024a3661d3858c8d2a40d48009ad9589d
Orig-Change-Id: Ic4176fc6ea3c118ae4570fdaf76ea39b192a7930
Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
Reviewed-on: http://android.intel.com:8080/75917
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Sun, Jianhua <jianhua.sun@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 Fat.cpp |   10 ++++------
 1 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/Fat.cpp b/Fat.cpp
index afdf00f..1c9c3eb 100644
--- a/Fat.cpp
+++ b/Fat.cpp
@@ -178,20 +178,18 @@ int Fat::format(const char *fsPath, unsigned int numSectors) {
     args[2] = "32";
     args[3] = "-O";
     args[4] = "android";
-    args[5] = "-c";
-    args[6] = "8";
 
     if (numSectors) {
         char tmp[32];
         snprintf(tmp, sizeof(tmp), "%u", numSectors);
         const char *size = tmp;
-        args[7] = "-s";
-        args[8] = size;
-        args[9] = fsPath;
+        args[5] = "-s";
+        args[6] = size;
+        args[7] = fsPath;
         rc = android_fork_execvp(ARRAY_SIZE(args), (char **)args, &status,
                 false, true);
     } else {
-        args[7] = fsPath;
+        args[5] = fsPath;
         rc = android_fork_execvp(8, (char **)args, &status, false,
                 true);
     }
-- 
1.7.4.1

